.github/workflows/commits.yml

```diff
@@ -0,0 +1,33 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: Conventional commit check
+on: [push]
+jobs:
+  commit_messages:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Load scopes from file
+        id: load_scopes
+        run: |
+          scopes_file=".github/.scopes.txt"
+          scopes=$(awk 'NR == 1 { printf $1 } NR > 1 { printf "|" $1 }' $scopes_file)
+          echo "::set-output name=scopes::$scopes"
+
+      - name: Conventional commit check
+        id: conventional_commit_check
+        run: |
+          regex="^(build|feat|fix|test|perf|refactor|docs|style)(\(($scopes)\))?:\s.*(\n\n.*)*(\n\n(\w|[-/])+.*)*$"
+          IFS='\'
+          commits=$(git log --pretty=format:"%s" $(git merge-base HEAD origin/main)..HEAD)
+          echo $commits
+          for commit in $commits; do
+            if ! [[ $commit =~ $regex ]]; then
+              echo "Invalid commit message: $commit"
+              exit 1
+            fi
+          done
```

Comment:
I was checking the output of the job, and for some reason, the error message was messed up:
```
fixup! docs: update contributing.md
feat: conventional commit check
docs: update contributing.md
Invalid commit message: fixup! docs: update contributing.md
feat: conventional commit check
docs: update contributing.md
```
https://github.com/cartesi/rollups/actions/runs/4961673199/jobs/8878760891#step:4:20

Reply:
So, debugging a little bit here, I found that `$commit` == `$commits`.